labgrid/driver/usbloader.py

```diff
@@ -82,3 +83,55 @@ def load(self, filename=None):
             self.loader.command_prefix +
             [self.tool, "-p", str(self.loader.path), "-c", mf.get_remote_path()]
         )
+
+
+@target_factory.reg_driver
+@attr.s(cmp=False)
+class RKUSBDriver(Driver, BootstrapProtocol):
+    bindings = {
+        "loader": {RKUSBLoader, NetworkRKUSBLoader},
+    }
+
+    image = attr.ib(default=None)
+    usb_loader = attr.ib(default=None)
+
+    def __attrs_post_init__(self):
+        super().__attrs_post_init__()
+        # FIXME make sure we always have an environment or config
+        if self.target.env:
+            self.tool = self.target.env.config.get_tool('rk-usb-loader') or 'rk-usb-loader'
+        else:
+            self.tool = 'rk-usb-loader'
+
+    def on_activate(self):
+        pass
+
+    def on_deactivate(self):
+        pass
+
+    @Driver.check_active
+    @step(args=['filename'])
+    def load(self, filename=None):
+        if self.target.env:
+            usb_loader = self.target.env.config.get_image_path(self.usb_loader)
+            mf = ManagedFile(usb_loader, self.loader)
+            mf.sync_to_resource()
+
+        subprocess.check_call(
+            self.loader.command_prefix
+            + [self.tool, 'db', mf.get_remote_path()]
+        )
+
+        if filename is None and self.image is not None:
+            filename = self.target.env.config.get_image_path(self.image)
+        mf = ManagedFile(filename, self.loader)
+        mf.sync_to_resource()
+
+        # wait the loader finish the process
+        time.sleep(1)
```

Comment:
Why is this sleep necessary here?

Reply:
We can confirm if the `sleep` is required. Worse, we can do a busy loop, trying to load it with a max of 3s for timeout.